Dragonflight/WarriorArms.lua

```diff
@@ -481,14 +510,16 @@ spec:RegisterUnitEvent( "UNIT_POWER_FREQUENT", "player", nil, function( event, u
     if powerType == "RAGE" then
         local current = UnitPower( "player", RAGE )
 
-        if current < lastRage then
-            rageSpent = ( rageSpent + lastRage - current ) % 20 -- Anger Mgmt.
+        if current < lastRage - 3 then -- Spent Rage, -3 is used as a Hack to avoid Rage decaying
 
-            if state.legendary.glory.enabled and FindPlayerAuraByID( 324143 ) then
-                gloryRage = ( gloryRage + lastRage - current ) % 20 -- Glory.
+            if state.talent.anger_management.enabled then
+                rageSpent = ( rageSpent + lastRage - current ) % 20
             end
-        end
 
+            if state.legendary.glory.enabled and state.buff.conquerors_banner.up then
```

Comment:
This part is more complicated.  Since we're reacting to an actual event here, we don't want to rely on `state.buff.conquerors_banner.up` because we have no idea what the virtual/forecasted game state looks like.  It could be working on predictions for several seconds in the future, and could've virtually expired Conqueror's Banner.  That's why FindPlayerAuraByID was used here; it is using real, current data rather than anything that is confabulated.

Reply:
Fixed in https://github.com/Hekili/hekili/pull/1670/commits/cfa753e27e69cd7db9d22ea72be7be411b29b541 - All specs go back to using FindPlayerAuraByID for Conq Buff being up.